3-CustomWebUI/3-1-CustomHTML/Console-Interactive-CustomWebUI/Program.cs

```diff
@@ -0,0 +1,137 @@
+        private static SystemWebViewOptions _customWebView = GetCustomHTML();
+
+        static async Task Main(string[] args)
+        {
+            // Using appsettings.json as our configuration settings
+            var builder = new ConfigurationBuilder()
+                .SetBasePath(System.IO.Directory.GetCurrentDirectory())
+                .AddJsonFile("appsettings.json");
+
+            configuration = builder.Build();
+
+            appConfiguration = configuration
+                .Get<PublicClientApplicationOptions>();
+
+            _authority = string.Concat(appConfiguration.Instance, appConfiguration.TenantId);
+
+            // Building a public client application
+            var app = PublicClientApplicationBuilder.Create(appConfiguration.ClientId)
+                                                    .WithAuthority(_authority)
+                                                    .WithRedirectUri(appConfiguration.RedirectUri)
+                                                    .Build();
+
+            string[] scopes = new[] { "user.read" };
+            AuthenticationResult result;
+
+            try
+            {
+                var accounts = await app.GetAccountsAsync();
+
+                // Try to acquire an access token from the cache. If an interaction is required, 
+                // MsalUiRequiredException will be thrown.
+                result = await app.AcquireTokenSilent(scopes, accounts.FirstOrDefault())
+                            .ExecuteAsync();
+            }
+            catch (MsalUiRequiredException)
+            {
+                // Acquiring an access token interactively. MSAL will cache it so we can use AcquireTokenSilent
+                // on future calls.
+                result = await app.AcquireTokenInteractive(scopes)
+                            .WithSystemWebViewOptions(_customWebView) // Using the custom html
+                            .ExecuteAsync();
+            }
+
+            string graphApiUrl = configuration.GetValue<string>("GraphApiUrl");
+            // Instantiating GraphServiceClient and using the access token acquired above.
+            var graphClient = GetGraphServiceClient(result.AccessToken, graphApiUrl);
+
+            // Calling the /me endpoint
+            var me = await graphClient.Me.Request().GetAsync();
+
+            // Printing the results
+            Console.Write(Environment.NewLine);
+            Console.WriteLine($"Hello {result.Account.Username}");
+            Console.Write(Environment.NewLine);
+            Console.WriteLine("-------- GRAPH RESULT --------");
+            Console.Write(Environment.NewLine);
+            Console.WriteLine($"Id: {me.Id}");
+            Console.WriteLine($"Display Name: {me.DisplayName}");
+            Console.WriteLine($"Email: {me.Mail}");
+
+            Console.ReadKey();
+        }
+
+        private static GraphServiceClient GetGraphServiceClient(string accessToken, string graphApiUrl)
```

Comment:
Could we use the pattern where the delegate does the try/catch?
like in https://github.com/Azure-Samples/ms-identity-dotnet-desktop-tutorial/pull/9